### DISCUSSION TITLE
Noob with a few quick questions

### DISCUSSION THREAD

**Discussion by cjn566** [jgromes/RadioLib#984] [Q&A] [answered]
Hi! I'm working on an art project that will include a radio component. I don't know nothing about radio tech, and before I go drowning myself in datasheets and acronyms, I was just wondering if someone can get me pointed in the right direction.

Basically, I bought a pair of the [these transcievers](https://www.digikey.com/en/products/detail/seeed-technology-co-ltd/109990166/7033232) after doing basically no research, and I want to if they will work for my purposes and also work with this library.

For my application:
- Power consumption is not an issue, power is supplied at for ends
- Transmission range is ~100-200m
- Data volume should be low, but having low latency is a priority

Basically I want to stream MIDI data from a keyboard, so for every keypress I need to get a packet with a few integers across before a noticeable delay.

After skimming the datasheet, it seems apparent that I should use the FSK/OOK modem instead of LoRa. Is that true? Will it work? Will it work with RadioLib?

Thanks in advance!

**Comment by jgromes** [comment, marked as answer]
The transceivers you linked are some RFM9x (probably RFM95/96, based on the listed 868 MHz frequency), and yes, RadioLib does support those. RFM9x are a copy of SX127x, so refer to those examples for details on usage.

As for your use case, I think it should be doable, but it depends on how strict is the latency requirement. A few tens of ms should be fine even when using a high-rate LoRa (e.g. LoRa pacekt with spreading factor 8, 250 kHz bandwidth and 4 data bytes will have airtime about 25 ms - at that point processing on both sides might introduce higher delay than the packet time-on-air, so make sure you have sufficiently powerful microcontroller). If that doesn't work for you, then FSK could be the way to go. Even on low data rates it is faster than LoRa, at the expense of decreased range. It also depends on the environment this will be set up in. 200m line-of-sight is quite a different story than 200m in a forest and over a hill.

**Comment by cjn566** [comment]
Amazing! That helps a lot.

I'm using a teensy 4 (600MHz) on both ends with almost no other processing, so should be pretty quick. I think <150ms start to finish would be a great target. And yes, line-of-sight is a guarantee for this project.

One thing I'd be interested in is having some degree of certainty that the receiver received, either with an ACK, or just having pretty bulletproof setup.. and I could just go do my own research at this point..

So knowing these things, would it make more sense / be easier to go with LoRa or with FSK for this project?

**Reply by jgromes** [reply]
> One thing I'd be interested in is having some degree of certainty that the receiver received, either with an ACK, or just having pretty bulletproof setup.

Before trying to do that, you have to ask yourself what are you going to do if that ACK does not arrive. Will you try to send the data again? And if not (for instance because another packet is already near transmission), is that information useful at all? I think I would only implement some ACK/retransmit mechanism in this scenario if I have enough time to repeat the whole tranmission again - and at that point, why not just always send two packets? ACK/retransmit is useful if you need to make sure the complete information arrives, and it always needs to be more efficient than sending redundant information.

> So knowing these things, would it make more sense / be easier to go with LoRa or with FSK for this project?

LoRa is a lot more forgiving when it comes to link margin, interference and frequency error between the transmitter and the receiver. I would start with a high LoRa data rate - something like SF8 BW250 could be a good start. Then, I would measure the SNR (signal-to-noise ratio) at the receiver, RadioLib has methods to get thet info from the LoRa module. If the SNR is sufficiently high (i.e., more then 3 dB over the demodulator limit, which is a function of the LoRa spreading factor), I would increase the speed. If it's too low or if teh link is unrelaible, I would decrease the speed. This way you should be able to find setting that has reliable link at the highest possible data rate.

**Reply by cjn566** [reply]
This is all really great information, I'm so glad I posted here. Thank you!
